src/components/stopwatch/StopWatchController.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <FreeRTOS.h>
+#include <optional>
+#include <timers.h>
+#include "utility/CircularBuffer.h"
+
+namespace Pinetime {
+  namespace System {
+    class SystemTask;
+  }
+  namespace Controllers {
+
+    enum class StopWatchStates { Cleared, Running, Paused };
+
+    struct LapInfo {
+      int count = 0;       // Used to label the lap
```

Comment:
Well, if we're talking naming, the entire concept of "laps" is a bit iffy here. Let's say I'm timing a hamster racing around a table-top track and it runs each lap in about 5 seconds. I'd expect to see something like
```
#1   00:05.02
#2   00:04.97
#3   00:04.99
```
but StopWatch shows me
```
#1   00:05.02
#2   00:09.99
#3   00:14.98
```
Rather than lap times, these are _split times_ (known from downhill skiing, for example). It's actually smarter to store split times: while L(n) = S(n) - S(n-1) hence O(1), S(n) = sum(L(1)..L(n)) hence O(n).

What's the proper naming, then? We could go abstract:
```
struct Measurement {
  int index = 0;
  TickType_t value = 0;
}
...
Utility::CircularBuffer<Measurement, histSize> history;
```
This would also be somewhat consistent with `MotionController`, the other user of `CircularBuffer`. Related fields and methods in StopWatch would have to adopt this terminology as well, of course.

I understand naming is important and hard to get right, so if you have better ideas I'm all ears. I know I have a bias towards abstract names but this is honestly the best I can come up with at the moment.

Reply:
I tried it out and abstract terminology is not a good fit for the public methods. How about this?
```
struct LapInfo {
  int number; // Used for labeling
  TickType_t timeSinceStart; // Excluding pauses
}
...
void AddLapToHistory();
int getMaxLapNumber();
std::optional<LapInfo> getLapFromHistory(int index); // 0 is most recent
...
static constexpr int histSize = 2;
Utility::CircularBuffer<LapInfo, histSize> history;
int maxLapNumber;
```

Reply:
Looks perfect to me, like that naming a lot 👍

Reply:
Applied.